ui/effects/use-screensize.js

```diff
@@ -1,18 +1,23 @@
 // Widths are taken from "ui/scss/init/vars.scss"
 import React from 'react';
 
-function useWindowSize() {
+export function useWindowSize(fn) {
   const isWindowClient = typeof window === 'object';
   const [windowSize, setWindowSize] = React.useState(isWindowClient ? window.innerWidth : undefined);
 
   React.useEffect(() => {
     function setSize() {
-      setWindowSize(window.innerWidth);
+      if (fn) {
+        const curr = fn(window.innerWidth);
+        if (windowSize !== curr) setWindowSize(curr);
```

Comment:
I made a small writeup for this (here)[https://github.com/OdyseeTeam/odysee-frontend/discussions/784] but TL;dr if fn is passed it will only-re render if the result of fn changes eg isMobile goes from true to false or vice versa. if fn is not passed for every resize event it will trigger one render. (you can see a demo of how much of a difference it makes in the writeup above)

Reply:
I think the confusion is this line: (windowSize !== curr) 

Windowsize is an integer, curr is a boolean?

Reply:
it doesn't really matter in this regard as even in typescript the type of windowSize is something like `React.useState<number | ReturnType<fn>>` hence if fn is passed the type of windowSize will remain constant throughout its lifecycle. with that said, maybe its the name windowSize, i could set it to something more generic like state but I think windowSize is still a good name as whats stored in it is still the windowSize like useIsMobile->useWindowSize->windowSize = boolean, while useWindowSize->windowSize = number. does that make sense?

Reply:
I see there was an issue with the initial state, I fixed that. hope that addresses this problem 

```
  const initialState = fn ? fn(window.innerWidth) : window.innerWidth;
  const [windowSize, setWindowSize] = React.useState(isWindowClient ? initialState : undefined);
```

Reply:
On the homepage, if you shrink the window width to invoke the Mobile layout, and then expand back, the sidebar goes missing.  If you then enter a File Page, it looks like this:
<img width="500" alt="image" src="https://user-images.githubusercontent.com/64950861/152579239-485aa083-b1de-4935-8d07-a58abc789e63.png">

Reply:
I can reproduce this issue, I'll fix it.

Reply:
Fixed, you can test again, ill post a writeup after dinner 🍜

Reply:
so `if (windowSize !== curr) setWindowSize(curr);` was the problem it basically there was a race condition that I didn't foresee, here windowSize was from useState and a few resize events triggered rapidly and lead to a situation where the state was reverted because of the multiple callbacks. 

the solution use a ref to store the current value of the state so the race condition is eliminated.

(In my defence this only happed if you used the quick screen widths of the dev tools, I manually resized so I didn't catch the bug 🙃 )

```js
  const prev = useRef();

  React.useEffect(() => {
    function setSize() {
      if (fn) {
        const curr = fn(window.innerWidth);
        if (prev !== curr) {
          setWindowSize(curr);
          prev.current = curr;
        }
      } else setWindowSize(window.innerWidth);
    }
```

Reply:
Actually, I manually resized too, but no worries, that's what PRs are for -- testing and review :)

The boolean vs integer overload is pretty a neat design, but I would still prefer to make things more obvious by splitting it up:  the `fn` version (renamed to something else?) will then just be a helper to `useIsMobile` and friends, while the integer version will be an actual `useWindowSize`.  

The cons is that both code will 95% similar, but at least the purposes will be obvious -- "give me current window size" vs. "tell me if size has crossed a threshold".

@tzarebczan, thoughts?

Reply:
I think it's worth it in this case, my help prevent confusion in the future even though its a DRY violstion. The code itself hopefully won't change too often.

Reply:
Done